File: taskhub_app/lib/models/proyecto.dart
import 'package:flutter/material.dart';

/// Una lista/proyecto que agrupa tareas (Trabajo, Personal...).
class Proyecto {
  final int id;
  final String nombre;
  final String colorHex; // "#RRGGBB"

  Proyecto({
    required this.id,
    required this.nombre,
    required this.colorHex,
  });

  factory Proyecto.fromJson(Map<String, dynamic> json) => Proyecto(
        id: json['id'] as int,
        nombre: json['nombre'] as String,
        colorHex: json['color'] as String? ?? '#4F46E5',
      );

  /// El color del proyecto como Color de Flutter.
  Color get color {
    final hex = colorHex.replaceAll('#', '');
    return Color(int.parse('FF$hex', radix: 16));
  }
}

File: TaskHub.API/Models/Proyecto.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace TaskHub.API.Models;

/// <summary>
/// Una lista o proyecto que agrupa tareas (ej: Trabajo, Personal).
/// </summary>
public class Proyecto
{
    public int Id { get; set; }

    [Required(ErrorMessage = "El nombre es obligatorio")]
    [MinLength(1)]
    public string Nombre { get; set; } = string.Empty;

    // Color en formato hex (#RRGGBB) para mostrar en la UI.
    public string Color { get; set; } = "#4F46E5";

    public DateTime CreadaEn { get; set; } = DateTime.UtcNow;

    // Dueño del proyecto.
    public int UserId { get; set; }

    [JsonIgnore]
    public User? User { get; set; }

    [JsonIgnore]
    public List<TaskItem> Tareas { get; set; } = new();
}
